### PR TITLE
[SPARK-36388][SPARK-36386][PYTHON] Fix DataFrame groupby-rolling and groupby-expanding to follow pandas 1.3

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to fix `RollingGroupBy` and `ExpandingGroupBy` to follow latest pandas behavior.

`RollingGroupBy` and `ExpandingGroupBy` no longer returns grouped-by column in values from pandas 1.3.

Before:
```python
>>> df = pd.DataFrame({"A": [1, 1, 2, 3], "B": [0, 1, 2, 3]})
>>> df.groupby("A").rolling(2).sum()
       A    B
A
1 0  NaN  NaN
  1  2.0  1.0
2 2  NaN  NaN
3 3  NaN  NaN
```

After:
```python
>>> df = pd.DataFrame({"A": [1, 1, 2, 3], "B": [0, 1, 2, 3]})
>>> df.groupby("A").rolling(2).sum()
       B
A
1 0  NaN
  1  1.0
2 2  NaN
3 3  NaN
```

### Why are the changes needed?

We should follow the behavior of pandas as much as possible.

### Does this PR introduce _any_ user-facing change?

Yes, the result of `RollingGroupBy` and `ExpandingGroupBy` is changed as described above.

### How was this patch tested?

Unit tests.

### CODE REVIEW

**Comment by itholic** [comment]
FYI: both SPARK-36388 and SPARK-36386 are resolved with same fix, so I consolidate them in this PR.

Also cc @ueshin @HyukjinKwon @xinrong-databricks

File: python/pyspark/pandas/window.py [hunk 1]
```diff
@@ -706,11 +706,16 @@ def _apply_as_series_or_frame(self, func: Callable[[Column], Column]) -> FrameLi
         if groupby._agg_columns_selected:
             agg_columns = groupby._agg_columns
         else:
-            agg_columns = [
-                psdf._psser_for(label)
+            agg_column_labels = [
+                label
                 for label in psdf._internal.column_labels
                 if label not in groupby._column_labels_to_exlcude
```
Comment (HyukjinKwon) [inline python/pyspark/pandas/window.py:712]: Can we add the column label to `column_labels_to_exlcude` (with some comments to explain why) at `DataFrameGroupBy` constructor?

Comment (itholic) [reply python/pyspark/pandas/window.py:712]: Got it, and seems like we should also rename `exlcude` to `exclude` ??

It's typo 😮

File: python/pyspark/pandas/window.py [hunk 2]
```diff
@@ -706,6 +706,10 @@ def _apply_as_series_or_frame(self, func: Callable[[Column], Column]) -> FrameLi
         if groupby._agg_columns_selected:
             agg_columns = groupby._agg_columns
         else:
+            # pandas doesn't keep the groupkey as a column from 1.3 for DataFrameGroupBy
+            if isinstance(groupby, DataFrameGroupBy):
+                for groupkey in groupby._groupkeys:  # type: ignore
+                    groupby._column_labels_to_exlcude.add(groupkey._internal.column_labels[0])
```
Comment (itholic) [inline python/pyspark/pandas/window.py:712]: Should we rename the from `exlcude` to `exclude` in this PR, too. ?? It's typo.

Comment (HyukjinKwon) [reply python/pyspark/pandas/window.py:712]: yeah please go ahead.

**Review by HyukjinKwon — APPROVED** [review]
Looks good. @ueshin would you mind taking a second look?

**Comment by HyukjinKwon** [comment]
@itholic can you check the test failures? https://github.com/itholic/spark/runs/3276195042?check_suite_focus=true

**Review by xinrong-meng — APPROVED** [review]

File: python/pyspark/pandas/window.py [hunk 3]
```diff
@@ -706,10 +706,16 @@ def _apply_as_series_or_frame(self, func: Callable[[Column], Column]) -> FrameLi
         if groupby._agg_columns_selected:
             agg_columns = groupby._agg_columns
         else:
+            # pandas doesn't keep the groupkey as a column from 1.3 for DataFrameGroupBy
+            if isinstance(groupby, DataFrameGroupBy):
+                for groupkey in groupby._groupkeys:  # type: ignore
+                    groupby._column_labels_to_exclude.add(  # type: ignore
```
Comment (ueshin) [inline python/pyspark/pandas/window.py:712]: Shall we not update `groupby._column_labels_to_exclude` directly?

```py
column_labels_to_exclude = groupby._column_labels_to_exclude.copy()
if isinstance(groupby, DataFrameGroupBy):
    for ...
        column_labels_to_exclude.add( ... )
agg_columns = [ ... if label not in column_labels_to_exclude]
```

**Review by ueshin — COMMENTED** [review]
Otherwise, LGTM.

File: python/pyspark/pandas/window.py (hunk 3 above)
Comment (itholic) [reply python/pyspark/pandas/window.py:712]: Oh, I got it. Let me address

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by ueshin** [comment]
@itholic We were missing to modify tests in `python/pyspark/pandas/tests/test_ops_on_diff_frames_groupby_expanding.py` and `python/pyspark/pandas/tests/test_ops_on_diff_frames_groupby_rolling.py`.
Could you submit a follow-up PR to fix them?
Thanks!

**Pull request closed without merging**